aspnetcore/security/blazor/server-side.md

```diff
@@ -0,0 +1,378 @@
+
+Blazor server-side apps offer a similar programming model to other UI frameworks for stateful apps, such as WPF, Windows Forms, or Blazor client-side. The main difference is that in several of the UI frameworks the memory consumed by the app belongs to the client and only affects that individual client. For example, a Blazor client-side app runs entirely on the client and only uses client memory resources. In the Blazor server-side scenario, the memory consumed by the app belongs to the server and is shared among clients on the server instance.
+
+Server-side memory demands are a consideration for all server-side apps. However, most web apps are stateless, and the memory used while processing a request is released when the response is returned. As a general recommendation, don't permit clients to allocate an unbound amount of memory as in any other server-side app that persists client connections. The memory consumed by a Blazor server-side app persists for a longer time than a single request.
+
+> [!NOTE]
+> During development, a profiler can be used or a trace captured to assess memory demands of clients. A profiler or trace won't capture the memory allocated to a specific client. To capture the memory use of a specific client during development, capture a dump and examine the memory demand of all the objects rooted at a user's circuit.
+
+### Client connections
+
+Connection exhaustion can occur when one or more clients open too many concurrent connections to the server, preventing other clients from establishing new connections.
+
+Blazor clients establish a single connection per session and keep the connection open for as long as the browser window is open. The demands on the server of maintaining all of the connections isn't specific to Blazor apps. Given the persistent nature of the connections and the stateful nature of Blazor server-side apps, connection exhaustion is a greater risk to availability of the app.
+
+<!--
+
+For ...
+
+> Blazor clients establish a single connection per session and keep the connection open for as long as the browser window is open.
+
+It seems like the word "connection" is overloaded. The topic speaks earlier to multiple connections per circuit, but here there is only one "connection" per user (circuit). I also wonder about "browser window" versus "broswer tab" in this context. Should this read something like ...
+
+> Blazor clients can establish multiple connections per user (circuit) and keep the connections open for as long as the browser window is open.
```

Comment:
The existing text seems fine to me. I agree "connection" is a bit overloaded here, but I think the main point still comes across.